src/main/java/uk/gov/pay/products/validations/RequestValidations.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.pay.products.validations;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.math.NumberUtils.isDigits;
+
+public class RequestValidations {
+
+    public Optional<List<String>> checkIsNumeric(JsonNode payload, String... fieldNames) {
+        return applyCheck(payload, isNotNumeric(), fieldNames, "Field [%s] must be a number");
+    }
+
+    public Optional<List<String>> checkIfExists(JsonNode payload, String... fieldNames) {
+        return applyCheck(payload, notExist(), fieldNames, "Field [%s] is required");
+    }
+
+    public Optional<List<String>> checkMaxLength(JsonNode payload, int maxLength, String... fieldNames) {
+        return applyCheck(payload, exceedsMaxLength(maxLength), fieldNames, "Field [%s] must have a maximum length of " + maxLength + " characters");
+    }
+
+    private Function<JsonNode, Boolean> exceedsMaxLength(int maxLength) {
+        return jsonNode -> jsonNode.asText().length() > maxLength;
+    }
+
+    public Optional<List<String>> applyCheck(JsonNode payload, Function<JsonNode, Boolean> check, String[] fieldNames, String errorMessage) {
+        List<String> errors = newArrayList();
```

Comment:
This api is internal isn't? I would encourage failfast rather than validate all and aggregate them.

Reply:
This is common functionality that we use in other microservices as well. The request validations can go into a commons package that we can pull from Maven and reuse it. I would rather keep it as it is and later on replace with the commons package.

Reply:
I meant changing it, but fine for now since that probably would take discussions with wider audience and agree to the approach.